spread/lxd.go

```diff
@@ -128,8 +154,18 @@ func (p *lxdProvider) Allocate(ctx context.Context, system *System) (Server, err
 		system: system,
 	}
 
-	printf("Waiting for lxd container %s to have an address...", name)
-	timeout := time.After(60 * time.Second)
+	what := "container"
+	if p.vm {
+		what = "VM"
+	}
+	printf("Waiting for lxd %s %s to have an address...", what, name)
+	maxTimeout := 60 * time.Second
+	if p.vm {
+		// VM may take considerably longer to start
+		// TODO: should this be configurable?
+		maxTimeout = 180 * time.Second
```

Comment:
I did not see this in my testing but it is probably true, are you using emulated VMs?

After the VM has an address, subsequent calls to `lxc exec` will fail until the `lxd-agent` inside the VM has started, https://github.com/canonical/spread/pull/221/commits/ab8232d56b5875f5e308ab91340e696a44e7a9ec could be useful for this.